[GTK][WPE] Add WPE layout testers for the EWS and two more bots for GTK API test queue

clopez opened the pull request:
#### a61b5a754b582534732e57437e8d22efb71fea4f
<pre>
[GTK][WPE] Add WPE layout testers for the EWS and two more bots for GTK API test queue
<a href="https://bugs.webkit.org/show_bug.cgi?id=250706">https://bugs.webkit.org/show_bug.cgi?id=250706</a>

Reviewed by Aakash Jain.

We would like to add WPE EWS layout testers for the EWS (new queue) and also two
more testers to the GTK API queue that sometimes is a bit slow.

This are the changes in this patch for the EWS buildbot config:
  - Rename queue &apos;WPE-EWS&apos; to &apos;WPE-Build-EWS&apos;
  - And a new queue named: &apos;WPE-WK2-Tests-EWS&apos; (EWS layout tests) with 8 workers.
  - Add 2 new workers for queue: API-Tests-GTK-EWS
  - Add 1 new worker for queue: WPE-Build-EWS

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(BuildFactory.__init__):
(WPEBuildFactory):
(WPETestsFactory):
(WPEFactory): Deleted.
(WPEFactory.__init__): Deleted.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/loadConfig.py:
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance.test_queues_without_relevance_info):

Canonical link: <a href="https://commits.webkit.org/260554@main">https://commits.webkit.org/260554@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9bb8d8fee9b96b39205e976e6f3348b01582a8d0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107314 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16371 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116470 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115914 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111213 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17797 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7609 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99470 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113078 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13435 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96653 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41045 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95339 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28279 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82814 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9402 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29632 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10046 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6485 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/106341 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15555 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49213 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11600 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3969 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/CISupport/ews-build/factories_unittest.py:
```diff
@@ -349,6 +349,26 @@ class TestExpectedBuildSteps(unittest.TestCase):
             'validate-change',
             'compile-webkit'
         ],
+        'WPE-WK2-Tests-EWS': [
```
We need `Tools/CISupport/ews-app` changes too, I think, is that right @aj062?

aj062 replied:
Yes. Although we can do that later (to add status-bubble on github), once the queues are running in the buildbot for few days smoothly without much failures/issues.

clopez replied:
Sure. I'm fine with having the bots running a few days on the builbot instance to ensure they run stable and as expected before adding the checkboxes to the github PRs.

clopez replied:
ping?
I would like to get this reviewed and landed, it has been here for 1 month already waiting and the new machines are mostly idle waiting to start running the tests.

clopez committed:
[GTK][WPE] Add WPE layout testers for the EWS and two more bots for GTK API test queue
https://bugs.webkit.org/show_bug.cgi?id=250706

Reviewed by Aakash Jain.

We would like to add WPE EWS layout testers for the EWS (new queue) and also two
more testers to the GTK API queue that sometimes is a bit slow.

This are the changes in this patch for the EWS buildbot config:
  - Rename queue 'WPE-EWS' to 'WPE-Build-EWS'
  - And a new queue named: 'WPE-WK2-Tests-EWS' (EWS layout tests) with 8 workers.
  - Add 2 new workers for queue: API-Tests-GTK-EWS
  - Add 1 new worker for queue: WPE-Build-EWS

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(BuildFactory.__init__):
(WPEBuildFactory):
(WPETestsFactory):
(WPEFactory): Deleted.
(WPEFactory.__init__): Deleted.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/loadConfig.py:
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance.test_queues_without_relevance_info):

Canonical link: https://commits.webkit.org/260554@main
